### PR TITLE
Add C# 9 test cases (8 rules)

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to #3668

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp9.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+﻿using System;
+
+int a = 0; // Compliant - FN
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp9.cs:3]: this is a local variable in the `main` method - why is this an FN?

same for line 8

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedReturnValue.CSharp9.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+SomeGenericMethod<object>();
+var o = SomeGenericMethod2<object>();
+
+T SomeGenericMethod2<T>() where T : class { return null; }
+T SomeGenericMethod<T>() where T : class { return null; } // Compliant
+
+new R().MyMethod();
+i = new R().MyMethod2();
+
+int NeverUsed(int neverUsedVal) // FN - the returned value is not used
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedReturnValue.CSharp9.cs:18]: does this work for local functions in C# 7?

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM, I've left 2 comments

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp9.cs (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp9.cs:3]: My bad. It's not.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedReturnValue.CSharp9.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedReturnValue.CSharp9.cs:18]: Yes, we have a few test cases with local functions and static local functions and the analysis works as expected. It seems we have issues only with local functions defined as top level statements.

**Pull request merged**